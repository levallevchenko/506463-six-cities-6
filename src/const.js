export const SortingTypes = {
  POPULAR: `Popular`,
  PRICE_LOW: `Price: low to high`,
  PRICE_HIGH: `Price: high to low`,
  RATING: `Top rated first`
};

export const CityNames = {
  PARIS: `Paris`,
  COLOGONE: `Cologne`,
  BRUSSELS: `Brussels`,
  AMSTERDAM: `Amsterdam`,
  HAMBURG: `Hamburg`,
  DUSSELDORF: `Dusseldorf`
};

export const Coordinates = {
  [CityNames.PARIS]: {
    latitude: 48.85661,
    longitude: 2.351499,
    zoom: 13
  },
  [CityNames.COLOGNE]: {
    latitude: 50.938361,
    longitude: 6.959974,
    zoom: 13
  },
  [CityNames.BRUSSELS]: {
    latitude: 50.846557,
    longitude: 4.351697,
    zoom: 13
  },
  [CityNames.AMSTERDAM]: {
    latitude: 52.37454,
    longitude: 4.897976,
    zoom: 13
  },
  [CityNames.HAMBURG]: {
    latitude: 53.550341,
    longitude: 10.000654,
    zoom: 13
  },
  [CityNames.DUSSELDORF]: {
    latitude: 51.225402,
    longitude: 6.776314,
    zoom: 13
  }
};

export const CardTypes = {
  MAIN: `MAIN`,
  FAVORITE: `FAVORITE`
};
