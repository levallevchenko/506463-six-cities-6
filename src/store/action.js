export const ActionType = {
  CHANGE_CITY: `offers/changeCity`,
  CHANGE_SORTING: `offers/changeSorting`,
  LOAD_OFFERS: `offers/loadOffers`,
  LOAD_OFFER_DATA: `offers/loadOfferData`,
  LOAD_REVIEWS: `offers/loadReviews`,
  LOAD_NEARBY_OFFERS: `offers/loadNearbyOffers`,
  SET_COMMENT: `offers/setComment`,
  LOAD_COMMENT: `offers/loadComment`,
  SUBMIT_COMMENT: `offers/submitComment`,
  SET_NOT_FOUND_OFFER: `offers/setNotFoundOffer`,
  SET_OFFERS_LOADING_FAILED: `offers/setOffersLoadingFailed`,
  REQUIRED_AUTHORIZATION: `user/requiredAuthorization`,
  SET_AUTH_INFO: `user/login`,
  LOAD_AUTH_INFO: `user/loadAuthInfo`,
  REDIRECT_TO_ROUTE: `app/redirectToRoute`,
  SET_ERROR: `app/setError`,
};

export const ActionCreator = {
  changeCity: (city) => ({
    type: ActionType.CHANGE_CITY,
    payload: city
  }),
  changeSorting: (sorting) => ({
    type: ActionType.CHANGE_SORTING,
    payload: sorting
  }),
  loadOffers: (offers) => ({
    type: ActionType.LOAD_OFFERS,
    payload: offers,
  }),
  loadOfferData: (offer) => ({
    type: ActionType.LOAD_OFFER_DATA,
    payload: offer,
  }),
  loadReviews: (reviews) => ({
    type: ActionType.LOAD_REVIEWS,
    payload: reviews,
  }),
  loadNearbyOffers: (nearby) => ({
    type: ActionType.LOAD_NEARBY_OFFERS,
    payload: nearby,
  }),
  setComment: (comment) => ({
    type: ActionType.SET_COMMENT,
    payload: comment
  }),
  loadComment: (info) => ({
    type: ActionType.LOAD_COMMENT,
    payload: info
  }),
  submitComment: (info) => ({
    type: ActionType.SUBMIT_COMMENT,
    payload: info
  }),
  setNotFoundOffer: () => ({
    type: ActionType.SET_NOT_FOUND_OFFER,
  }),
  setOffersLoadingFailed: () => ({
    type: ActionType.SET_OFFERS_LOADING_FAILED,
  }),
  setAuthInfo: (info) => ({
    type: ActionType.SET_AUTH_INFO,
    payload: info
  }),
  loadAuthInfo: (info) => ({
    type: ActionType.LOAD_AUTH_INFO,
    payload: info
  }),
  requireAuthorization: (status) => ({
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status,
  }),
  redirectToRoute: (url) => ({
    type: ActionType.REDIRECT_TO_ROUTE,
    payload: url
  }),
  setError: (error) => ({
    type: ActionType.SET_ERROR,
    payload: error
  })
};
