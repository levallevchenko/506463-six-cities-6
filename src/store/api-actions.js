import {ActionCreator} from "./action";
import {AuthorizationStatus, APIRoute} from "../const";

export const fetchOffersList = () => (dispatch, _getState, api) => (
  api.get(APIRoute.OFFERS)
    .then(({data}) => dispatch(ActionCreator.loadOffers(data)))
);

export const fetchOffer = (id) => (dispatch, _getState, api) => (
  api.get(`/hotels/${id}`)
    .then(({data}) => dispatch(ActionCreator.loadOfferData(data)))
    .catch(() => dispatch(ActionCreator.setNotFoundOffer()))
);

export const fetchReviews = (id) => (dispatch, _getState, api) => (
  api.get(`/comments/${id}`)
    .then(({data}) => dispatch(ActionCreator.loadReviews(data)))
);

export const fetchNearbyOffers = (id) => (dispatch, _getState, api) => (
  api.get(`/hotels/${id}/nearby`)
    .then(({data}) => dispatch(ActionCreator.loadNearbyOffers(data)))
);

export const postComment = ({review: comment, rating}, id) => (dispatch, _getState, api) => {
  dispatch(ActionCreator.loadComment(true));
  api.post(`/comments/${id}`, {comment, rating})
    .then(({data}) => dispatch(ActionCreator.setComment(data)))
    .catch(() => {});
};

export const checkAuth = () => (dispatch, _getState, api) => (
  api.get(APIRoute.LOGIN)
    .then(({data}) => dispatch(ActionCreator.setAuthInfo(data)))
    .then(() => dispatch(ActionCreator.loadAuthInfo(true)))
    .then(() => dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH)))
    .catch(() => {})
);

export const login = ({login: email, password}) => (dispatch, _getState, api) => (
  api.post(APIRoute.LOGIN, {email, password})
    .then(({data}) => dispatch(ActionCreator.setAuthInfo(data)))
    .then(() => dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH)))
    .then(() => dispatch(ActionCreator.redirectToRoute(`/`)))
    .catch(() => {})
);
