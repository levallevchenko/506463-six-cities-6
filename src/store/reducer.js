import {CityName, SortingType, AuthorizationStatus} from '../const';
import {ActionType} from './action';
import {adaptOffersData} from '../services/adapter';

const initialState = {
  activeSorting: SortingType.POPULAR,
  activeCity: CityName.PARIS,
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  authInfo: {},
  isOffersLoaded: false,
  offers: [],
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_CITY:
      return {
        ...state,
        activeCity: action.payload
      };
    case ActionType.CHANGE_SORTING:
      return {
        ...state,
        activeSorting: action.payload
      };
    case ActionType.LOAD_OFFERS:
      return {
        ...state,
        offers: adaptOffersData(action.payload),
        isOffersLoaded: true
      };
    case ActionType.REQUIRED_AUTHORIZATION:
      return {
        ...state,
        authorizationStatus: action.payload
      };
    case ActionType.SET_AUTH_INFO:
      return {
        ...state,
        authInfo: action.payload
      };
    default:
      return state;
  }
};

export {reducer};
