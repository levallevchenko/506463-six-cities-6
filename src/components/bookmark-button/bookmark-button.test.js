import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {NameSpace} from '../../store/root-reducer';
import {AuthorizationStatus} from '../../const';
import BookmarkButton from './bookmark-button';

const mockStore = configureStore({});
it(`BookmarkButton should render correctly`, () => {
  const history = createMemoryHistory();

  const store = mockStore({
    [NameSpace.USER]: {
      authStatus: AuthorizationStatus.NO_AUTH,
      isAuthInfoLoaded: true,
    },
    [NameSpace.OFFERS]: {
      favoriteOffers: [{}],
    },
  });

  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <BookmarkButton markupBlock={`test__`} hotelId={1} />
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
