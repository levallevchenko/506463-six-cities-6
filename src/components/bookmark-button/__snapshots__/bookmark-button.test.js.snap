// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkButton should render correctly 1`] = `
<div>
  <button
    class="test__bookmark-button button"
    type="button"
  >
    <svg
      class="test__bookmark-icon"
      height="19"
      width="18"
    >
      <use
        xlink:href="#icon-bookmark"
      />
    </svg>
    <span
      class="visually-hidden"
    >
      To bookmarks
    </span>
  </button>
</div>
`;
